### DISCUSSION TITLE
It is easy to get lost in the automation list when there are if/then or choice blocks [new user observation 5 of 7]

### DISCUSSION THREAD

**Discussion by davidbbs** [home-assistant/frontend#17230] [Automations / scripts / scenes]
I observed a new user creating simple light automations for three hours this weekend. 
Observation 5 of 7:

**Problem 5: It is easy to get lost in the automation list when there are if/then or choice blocks**
**Proposal 5: Color each depth level the same (and increasingly dark) shade**
_(The new users was constantly scrolling up and down to try and figure out where they were in if/then blocks and choose blocks. The worst was adding an OR block which made the if/then even harder to follow.)_

This can be solved by using increasingly dark shades of gray (or alternating) as background colors for each level of depth in the hierarchy. This would make sure everything in the “if" part was visually separated from the "then” part and that the items in the “or” part in the “if” part were visually grouped. (Clearly there are problems with how many levels deep and how dark, but doing this for just the first 4 levels would probably cover 99% of new users.)

**Comment by Sawtaytoes** [comment]
I need this. I keep getting lost in automations, and it's hard to remember what everything does. I have over 50 automations and scripts.

Here's one I've messed up twice already which controls my thermostats in Cool, Heat, and Heat/Cool modes:
![image](https://github.com/user-attachments/assets/a3f04abb-89ae-4eef-ad88-f950e465cf10)

What's crazy is Home Assistant already has a better UI to visualize automations, you just can't access it outside of the Trace view. That sucks! It means you have to run the automation to even see this view, and it's not like you can edit anything in here.

Here's one that loops back on itself until the lights turn off:
![image](https://github.com/user-attachments/assets/3f3cab98-30f0-4c45-b710-71626466ecad)

Imagine editing this in the Home Assistant 1-column view.

If it at least made multiple columns like this, I'd be able to edit it easier. The current UI is very phone-friendly, yet it's not because it's very difficult to drag things around and especially add entities or devices unless you turn your phone horizontally.

A view with these little dots would actually be more usable and mobile-friendly.

It doesn't have to be the only view, it just needs to be _there... somewhere_. Unlike the original proposal to change some colors, this view already exists, it's just that the "Step Config" only displays YAML, not editable blocks.
